### ISSUE TITLE
Nested parameter descriptions

### ISSUE THREAD

**Issue by jonchardy** [TypeStrong/typedoc#497]
TypeDoc doesn't seem to have a way to provide comments for nested parameters. Consider the following TypeScript method:

```js
/**
 * Test method with object param
 * @param {Object} options
 */
public foo(options: { first: string, second: number, third: (a: number) => number }) {

}
```

TypeDoc generates something like this:
![image](https://cloud.githubusercontent.com/assets/1202288/25919263/3dd611ca-359c-11e7-8539-e3a6f73ac6e2.png)

Ideally there would be a way to provide descriptions for all parts of the parameter (first, second, third, a, returned number) using some sort of syntax within the comment. As it stands, I suppose it's possible to provide some sort of long description for the options param which describes each component of it, but then it becomes messy having the descriptions separated from the type information.

**Comment by aciccarello** [comment]
@jonchardy The best way to document a parameter object like that would be to define an interface. That also allows you to to reuse the interface. Is there a particular reason why you wouldn't define the interface externally?

**Comment by jonchardy** [comment]
Yea, that's probably a better option. Definitely more TypeScript-y.

**Comment by aciccarello** [comment]
@jonchardy I was just looking at [one of the tests](https://github.com/TypeStrong/typedoc/blob/master/src/test/converter/decorators/decorators.ts#L42) and it appears that you may be able document these as well.

```TypeScript
/**
 * Test method with object param
 * @param options Describing the options param
 * @param options.first A description of the first property of the options param
 */
public foo(options: { first: string, second: number, third: (a: number) => number }) {

}
```

This matches [how JSDoc handles `@param`](http://usejsdoc.org/tags-param.html). It should be added to the TypeDoc docs.

**Comment by StephanBijzitter** [comment]
I'm a bit confused.

This issue is closed, but it's in a project with status "to do" and what I get today is even less than the screenshot in the opening post.
All I see is `options: object`.

What do I do?

**Comment by Gerrit0** [comment]
The project it is in is documentation - we need to create clearer documentation on how this feature works.

The `options: object` is another bug - #1103 provides a partial fix that has one fatal flaw, and it will be fixed with #801.

**Comment by StephanBijzitter** [comment]
Ah, thanks! With the information in those issues I got it working indeed :-)
Not perfect, but I'm happy.
